integration-tests/src/server.js

```diff
@@ -0,0 +1,64 @@
+const express = require('express')
+const http = require('http')
+const puppeteer = require('puppeteer')
+const _ = require('lodash/fp')
+const { promiseHandler, Response } = require('./utils')
+const { testFindWorkflow } = require('../tests/find-workflow')
+const { testImportCohortData } = require('../tests/import-cohort-data')
+const { testImportDockstoreWorkflow } = require('../tests/import-dockstore-workflow')
+const { testRegisterUser } = require('../tests/register-user')
+const { testRunWorkflow } = require('../tests/run-workflow')
+const { defaultTimeout } = require('../utils/integration-helpers')
+const { delay, withScreenshot } = require('../utils/integration-utils')
+const envs = require('../utils/terra-envs')
+
+
+const app = express()
+const server = http.createServer(app)
+server.setTimeout(defaultTimeout + 60 * 1000) // allow an extra minute for teardown
+
+const getBrowser = _.once(() => puppeteer.launch())
+const getContext = async () => {
+  const browser = await getBrowser()
+  return browser.createIncognitoBrowserContext()
+}
+
+const testTimeout = async (context, timeout) => {
+  await delay(timeout)
+  context.close()
+  throw new Error(`Test timeout after ${timeout}ms`)
+}
+
+const registerTestEndpoint = ({ fn, name, timeout = defaultTimeout }) => {
+  const path = `/test/${name}`
+  console.info(`=> ${path}`)
+  app.post(path, promiseHandler(async req => {
+    const targetEnvParams = { ...envs[req.query.targetEnv] }
+    const context = await getContext()
+    const page = await context.newPage()
+    try {
+      const result = await Promise.race([
+        withScreenshot(name)(fn)({ context, page, ...targetEnvParams }),
+        testTimeout(context, timeout)
+      ])
+      return new Response(200, result)
+    } catch (e) {
+      return new Response(200, JSON.stringify({
```

Comment:
nit: We don't need to stringify this manually, objects should get sent as JSON automatically.

Reply:
I'll experiment. I didn't get much of anything useful when including `e` in the response. `JSON.stringify` might have been from an earlier attempt to do that which ultimately led to me constructing this object that manually extracts `name`, `message`, and `stack` from the exception.